lib/resty/worker/events.lua

```diff
@@ -298,18 +311,18 @@ _M.poll = function()
     return nil, err
   end
 
-  local count = 0
+  local count = event_id - _last_event
   local cache_data = {}
   local cache_err = {}
   -- in case an event id has been published, but we're fetching it before
   -- its data was posted and we have to wait, we don't want the next
   -- event to timeout before we get to it, so go and cache what's
   -- available, to minimize lost data
-  while _last_event < event_id do
-    count = count + 1
+
+  for i = 1, count do
```

Comment:
this is not just style. Can we make this a separate commit? (I have to thorouglhy read it still to make sure there are no side-effects)

Reply:
fixed too